plugins/woocommerce/includes/class-wc-tracker.php

```diff
@@ -498,6 +498,66 @@ private static function get_order_dates() {
 		return array_merge( $min_max, $processing_min_max );
 	}
 
+	/**
+	 * Extract the group key for an associative array of objects which have unique ids in the key.
+	 * A 'group_key' property is introduced in the object.
+	 * For example, two objects with keys like 'WooDataPay ** #123' and 'WooDataPay ** #78' would
+	 * both have a group_key of 'WooDataPay **' after this function call.
+	 *
+	 * @param array  $objects     The array of objects that need to be grouped.
+	 * @param string $default_key The property that will be the default group_key.
+	 * @return array Contains the objects with a group_key property.
+	 */
+	private static function extract_group_key( $objects, $default_key ) {
+		$keys = array_keys( $objects );
+
+		// Sort keys by length and then by characters within the same length keys.
+		usort(
+			$keys,
+			function( $a, $b ) {
+				if ( strlen( $a ) === strlen( $b ) ) {
+					return strcmp( $a, $b );
+				}
+				return ( strlen( $a ) < strlen( $b ) ) ? -1 : 1;
+			}
+		);
+
+		// Look for common tokens in every pair of adjacent keys.
+		$prev = '';
+		foreach ( $keys as $key ) {
+			if ( $prev ) {
+				$comm_tokens = array();
+
+				// Tokenize the current and previous gateway names.
+				$curr_tokens = preg_split( '/[ :,\-_]/ ', $key );
+				$prev_tokens = preg_split( '/[ :,\-_]/ ', $prev );
+
+				$len_curr = count( $curr_tokens );
+				$len_prev = count( $prev_tokens );
+
+				// Gather the common tokens.
+				// Let us allow for the unique reference id to be anywhere in the name but for the first token.
```

Comment:
Is the "but for the first token" part of this comment accurate? I don't see where the first token is handled differently.

Reply:
That is a remnant of a logic I wrote earlier. I've removed it.